From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xiaoyin Liu <xiaoyinl@users.noreply.github.com>
Date: Sun, 28 May 2023 19:09:43 +0400
Subject: [PATCH] 5395: Close FileDataTypeManager in CompareGDTs.java

`firstArchive` and `secondArchive` are not closed, which causes errors the next time you run this script with the same GDTs, because the two GDT files are still in use.
---
 Ghidra/Features/Base/ghidra_scripts/CompareGDTs.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Ghidra/Features/Base/ghidra_scripts/CompareGDTs.java b/Ghidra/Features/Base/ghidra_scripts/CompareGDTs.java
index 55a2fb4e5..a2e399bd1 100644
--- a/Ghidra/Features/Base/ghidra_scripts/CompareGDTs.java
+++ b/Ghidra/Features/Base/ghidra_scripts/CompareGDTs.java
@@ -69,6 +69,7 @@ public class CompareGDTs extends GhidraScript {
 			popup(
 				"An architecture language error occured while opening archive (see log for details)\n" +
 					secondFile.getPath());
+			firstArchive.close();
 			return;
 		}
 
@@ -83,6 +84,8 @@ public class CompareGDTs extends GhidraScript {
 		}
 		finally {
 			printWriter.close();
+			firstArchive.close();
+			secondArchive.close();
 		}
 	}
 
-- 
2.40.1

